pkg/apis/serving/v1beta1/predictor_model.go

```diff
@@ -199,6 +198,22 @@ func sortClusterServingRuntimeList(runtimes *v1alpha1.ClusterServingRuntimeList)
 	})
 }
 
+func sortSupportedRuntimeByPriority(runtimes []v1alpha1.SupportedRuntime, modelFormat ModelFormat) {
+	sort.Slice(runtimes, func(i, j int) bool {
+		p1 := runtimes[i].Spec.GetPriority(modelFormat.Name)
+		p2 := runtimes[j].Spec.GetPriority(modelFormat.Name)
+
+		if p1 == nil && p2 == nil {
+			return false
+		} else if p1 == nil && p2 != nil {
```

Comment:
if priority is not specified it means lower priority right?

Reply:
Actually, It is affected by the sorting https://github.com/kserve/kserve/blob/32be7464b1a23a59c7980f91175c6c1ff35f4b0a/pkg/apis/serving/v1beta1/predictor_model.go#L100. serving runtime is sorted by created timestamp desc and name asc. So if both of the runtimes does not specify the priority then the order is maintained as it is. If any one of them specifies the priority then that takes precedence.